File: Backend/Backend/Models/DTOs/RequestSessionDto.cs
namespace Backend.Models.DTOs
{
    public class RequestSessionDto
    {
        public int Id { get; set; }
        public int GameId { get; set; }
        public int RemainingSeconds { get; set; }
        public bool IsExpired { get; set; }
        public int CurrentNumber { get; set; }
    }
}

File: frontend/src/types/RequestSessionDto.ts
import { Game } from "./Game";
import { Rule } from "./Rule";

// When a session starts or is queried
export interface RequestSessionDto {
  id: number;
  game: Game;
  startTime: string; // ISO string
  remainingSeconds: number;
  isExpired: boolean;
  currentNumber: number;
  totalCorrect: number;
  totalIncorrect: number;
  rules: Rule[];
}